src/Preg.php

```diff
@@ -87,12 +91,14 @@ public static function matchAll($pattern, $subject, &$matches = null, $flags = P
      */
     public static function replace($pattern, $replacement, $subject, $limit = -1, &$count = null)
     {
-        $result = @preg_replace(self::addUtf8Modifier($pattern), $replacement, $subject, $limit, $count);
+        $patterns = self::verifyPatterns((array) $pattern);
```

Comment:
if we don't provide an array, please not convert it to array. verifyPatterns can accept string|string[], like all other methods

Reply:
There was a split, verifyPatterns for arrays, verify for strings. I perhaps should more properly type-hint that.

Reply:
Was already type hinted. It's not a conversion but a cast. PHP is very good at this, I see it as less overhead and would really not suggest to change this.

Reply:
This method is now gone, however would you still suggest to not cast before calling the method but preferable doe that kind of processing inside the method?

Even though I assume your comments are addressed due to removal of the method, I'm interested in your opinion as there is a new method now which uses the same pattern.